### PR TITLE
switching compression algorithm to brotli

### PR DESCRIPTION
**Pull request by dlockhart**
The browser support is good for brotli, and it's a better algorithm than gzip so might as well use it.

| Page  | No Compression | GZIP | Brotli |
| ------------- | ------------- | ------------- | ------------- |
| Homepage  | 4.6 MB  | 757 kB | 563 kB |
| Component page  | 17 MB  | 4.1 MB | 3.0 MB |

So with Brotli, we're seeing 26%-37% smaller resources -- sweet!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,18 +43,18 @@ jobs:
...
         env:
           # cred variables set in the "Assume role" step
           AWS_DEFAULT_REGION: ca-central-1
       - name: Deploy - branch playground
         run: |
-          find ./playground-build/ \( -iname '*.html' -o -iname '*.css' -o -iname '*.js' -o -iname '*.json' -o -iname '*.svg' \) -exec gzip -9 -n {} \; -exec mv {}.gz {} \;
-          aws s3 sync --delete ./playground-build/ "s3://daylight.d2l.dev/branches/pr-${{ github.event.number }}-playground-build" --content-encoding 'gzip' --exclude '*.*' --include '*.html' --include '*.css' --include '*.json' --include '*.svg' --include 'playground-service-worker.js'
-          aws s3 sync --delete ./playground-build/ "s3://daylight.d2l.dev/branches/pr-${{ github.event.number }}-playground-build" --content-encoding 'gzip' --cache-control "public,max-age=31536000,immutable" --exclude '*.*' --include '*.js' --exclude 'playground-service-worker.js'
+          find ./playground-build/ \( -iname '*.html' -o -iname '*.css' -o -iname '*.js' -o -iname '*.json' -o -iname '*.svg' \) -exec brotli -9 -n {} \; -exec mv {}.br {} \;
```
Comment (omsmith) [inline .github/workflows/ci.yml:55]:
```suggestion
          find ./playground-build/ \( -iname '*.html' -o -iname '*.css' -o -iname '*.js' -o -iname '*.json' -o -iname '*.svg' \) -exec brotli -n {} \; -exec mv {}.br {} \;
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -48,18 +48,18 @@ jobs:
           aws-region: ca-central-1
       - name: Deploy - main
         run: |
-          find ./_site/ \( -iname '*.html' -o -iname '*.css' -o -iname '*.js' -o -iname '*.json' -o -iname '*.svg' \) -exec gzip -9 -n {} \; -exec mv {}.gz {} \;
-          aws s3 sync --delete ./_site/ s3://daylight.d2l.dev/branches/main --content-encoding 'gzip' --exclude '*.*' --include '*.html' --include '*.css' --include '*.json' --include '*.svg'
-          aws s3 sync --delete ./_site/ s3://daylight.d2l.dev/branches/main --content-encoding 'gzip' --cache-control "public,max-age=31536000,immutable" --exclude '*.*' --include '*.js'
+          find ./_site/ \( -iname '*.html' -o -iname '*.css' -o -iname '*.js' -o -iname '*.json' -o -iname '*.svg' \) -exec brotli -9 -n {} \; -exec mv {}.gz {} \;
```
Comment (omsmith) [inline .github/workflows/deploy.yml:51]:
```suggestion
          find ./_site/ \( -iname '*.html' -o -iname '*.css' -o -iname '*.js' -o -iname '*.json' -o -iname '*.svg' \) -exec brotli -n {} \; -exec mv {}.gz {} \;
```

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -48,18 +48,18 @@ jobs:
...
           aws s3 sync --delete ./_site/ s3://daylight.d2l.dev/branches/main --exclude '*.html' --exclude '*.css' --exclude '*.js' --exclude '*.json' --exclude '*.svg'
         env:
           # cred variables set in the "Assume role" step
           AWS_DEFAULT_REGION: ca-central-1
       - name: Deploy - playground elements
         run: |
-          find ./playground-build/ \( -iname '*.html' -o -iname '*.css' -o -iname '*.js' -o -iname '*.json' -o -iname '*.svg' \) -exec gzip -9 -n {} \; -exec mv {}.gz {} \;
-          aws s3 sync --delete ./playground-build/ "s3://daylight.d2l.dev/branches/main-playground-build" --content-encoding 'gzip' --exclude '*.*' --include '*.html' --include '*.css' --include '*.json' --include '*.svg'
-          aws s3 sync --delete ./playground-build/ "s3://daylight.d2l.dev/branches/main-playground-build" --content-encoding 'gzip' --cache-control "public,max-age=31536000,immutable" --exclude '*.*' --include '*.js' --exclude 'playground-service-worker.js'
+          find ./playground-build/ \( -iname '*.html' -o -iname '*.css' -o -iname '*.js' -o -iname '*.json' -o -iname '*.svg' \) -exec brotli -9 -n {} \; -exec mv {}.gz {} \;
```
Comment (omsmith) [inline .github/workflows/deploy.yml:60]:
```suggestion
          find ./playground-build/ \( -iname '*.html' -o -iname '*.css' -o -iname '*.js' -o -iname '*.json' -o -iname '*.svg' \) -exec brotli -n {} \; -exec mv {}.gz {} \;
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dlockhart) [reply .github/workflows/ci.yml:55]: My bad, should have removed that everywhere. Do we still want `-n`?

>  "Do not copy source file attributes"

Comment (omsmith) [reply .github/workflows/ci.yml]: :shrug:

**Review by oelhanafey — APPROVED** [review]
Nice, it looks like a significant improvement!

**Review by eKoopmans — APPROVED** [review]
🎉

**Pull request merged**